cmd/deploy/deploy.go

```diff
@@ -651,6 +656,27 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 	}
 }
 
+func (dc *DeployCommand) validateK8sResources(ctx context.Context, manifest *model.Manifest) error {
```

Comment:
NIT: I don't know if it's the command the one that has to validate it. I think that its external struct the one that should validate itself after setting it. WDYT

Reply:
I am using the interface that exposes the behavior of the external resources package. The same principle used to make use of the Deploy() and List() logic of an external resource. In this case, the deploy command only knows that it has to call Validate() for each external resource in the manifest

Reply:
What I mean is that I don't think that the deploy command should be the one that calls the validations. I think that it's the `GetManifest` function the one that should check that everything is correct.

We can leave it outside of this PR scope since it requires a big refactor of the function but if we create a new command that needs to validate the external resources we will need to remember that we need to call this new function that is not under the `deployCommand` scope